tests/scene/test_occluder_instance_3d.h

```diff
@@ -0,0 +1,127 @@
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_OCCLUDER_INSTANCE_3D_H
+#define TEST_OCCLUDER_INSTANCE_3D_H
+
+#include "core/math/vector3.h"
+#include "scene/3d/mesh_instance_3d.h"
+#include "scene/3d/occluder_instance_3d.h"
+#include "scene/main/scene_tree.h"
+#include "scene/resources/3d/primitive_meshes.h"
+
+#include "tests/test_macros.h"
+
+namespace TestOccluderInstance3D {
+
+TEST_CASE("[SceneTree][OccluderInstance3D] Test baking functionality") {
+// We need to load an occluder path because when the light occluder generates
+// it has to save to a file.
+#ifdef WINDOWS_ENABLED
+	const String occluder_path = OS::get_singleton()->get_environment("TEMP").path_join("test_occluder.occ");
+#else
+	const String occluder_path = "/tmp/test_occluder.occ";
+#endif
+
+	SceneTree *scene_tree = SceneTree::get_singleton();
+
+	// Create our occluder to do tests with.
+	OccluderInstance3D *occluder_instance = memnew(OccluderInstance3D);
+
+	// Root node to put meshes under.
+	Node *test_bake_scene = memnew(Node);
+
+	scene_tree->get_root()->add_child(test_bake_scene);
+	scene_tree->get_root()->add_child(occluder_instance);
+
+	// Instantiate a mesh to generate.
+	MeshInstance3D *box_mesh_instance = memnew(MeshInstance3D);
+	Ref<BoxMesh> box_mesh = Ref<BoxMesh>();
+	box_mesh.instantiate();
+	box_mesh_instance->set_mesh(box_mesh);
+	test_bake_scene->add_child(box_mesh_instance);
+	box_mesh_instance->set_owner(test_bake_scene);
+
+	// This is the vertices output that we expect for a single cube.
+	const PackedVector3Array expected_vertices_output = PackedVector3Array(
+			{ Vector3(-0.5, 0.5, 0.5), Vector3(0.5, 0.5, 0.5), Vector3(-0.5, -0.5, 0.5), Vector3(0.5, -0.5, 0.5), Vector3(0.5, 0.5, -0.5), Vector3(-0.5, 0.5, -0.5),
+					Vector3(0.5, -0.5, -0.5), Vector3(-0.5, -0.5, -0.5) });
```

Comment:
Reformat to improve readability:

```suggestion
	const PackedVector3Array expected_vertices_output = PackedVector3Array({
		Vector3(-0.5, 0.5, 0.5),
		Vector3(0.5, 0.5, 0.5),
		Vector3(-0.5, -0.5, 0.5),
		Vector3(0.5, -0.5, 0.5),
		Vector3(0.5, 0.5, -0.5),
		Vector3(-0.5, 0.5, -0.5),
		Vector3(0.5, -0.5, -0.5),
		Vector3(-0.5, -0.5, -0.5),
	});
```